service/geopmdpy_test/TestPlatformService.py

```diff
@@ -96,21 +99,525 @@ def test_get_group_access_default(self):
+        server_pid = 123
+
+        with mock.patch('geopmdpy.pio.stop_batch_server', return_value=[]) as sbd:
+            self._platform_service.stop_batch(client_pid, server_pid)
+            sbd.assert_called_once_with(server_pid)
+
+    def test_read_signal_invalid(self):
+        operation = 'PlatformReadSignal'
+        err_msg = "Operation '{}' not allowed without an open session".format(operation)
+        with self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.read_signal('', '', '', '')
+
+        session_data = self._write_session_files_helper()
+        client_pid = session_data['client_pid']
+        self._platform_service._sessions[client_pid] = session_data
+
+        signal_name = 'geopm'
+        err_msg = 'Requested signal that is not in allowed list: {}'.format(signal_name)
+        with self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.read_signal(client_pid, signal_name, '', '')
+
+    def test_read_signal(self):
+        session_data = self._write_session_files_helper()
+        client_pid = session_data['client_pid']
+        self._platform_service._sessions[client_pid] = session_data
+
+        signal_name = 'energy'
+        domain = 7
+        domain_idx = 42
+        with mock.patch('geopmdpy.pio.read_signal', return_value=[]) as rs:
+            self._platform_service.read_signal(client_pid, signal_name, domain, domain_idx)
+            rs.assert_called_once_with(signal_name, domain, domain_idx)
+
+    def test_write_control_invalid(self):
+        operation = 'PlatformWriteControl'
+        err_msg = "Operation '{}' not allowed without an open session".format(operation)
+        with self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.write_control('', '', '', '', '')
+
+        session_data = self._write_session_files_helper()
+        client_pid = session_data['client_pid']
+        self._platform_service._sessions[client_pid] = session_data
+
+        control_name = 'energy'
+        err_msg = 'Requested control that is not in allowed list: {}'.format(control_name)
+        with self.assertRaisesRegex(RuntimeError, err_msg):
+            self._platform_service.write_control(client_pid, control_name, '', '', '')
+
+    def test_write_control(self):
+        session_data = self._write_session_files_helper()
+        client_pid = session_data['client_pid']
+        self._platform_service._sessions[client_pid] = session_data
+
+        control_name = 'geopm'
+        domain = 7
+        domain_idx = 42
+        setting = 777
+        with mock.patch('geopmdpy.pio.write_control', return_value=[]) as wc, \
+             mock.patch('geopmdpy.pio.save_control', return_value=[]):
+            self._platform_service.write_control(client_pid, control_name, domain, domain_idx, setting)
+            wc.assert_called_once_with(control_name, domain, domain_idx, setting)
+
+    def test__read_allowed_invalid(self):
```

Comment:
Why the double underscore here and below?

Reply:
Because this test is for a private/internal method of the PlatformService: ```_read_allowed()```.

Thoughts?  I can drop it.

Reply:
This is probably tested already with the get_group_access_named test.  Check it.

Reply:
This is still needed to cover the exception out of _read_allowed.